### PR TITLE
Radix to Base UI migration

### PR DESCRIPTION
**Pull request by wwwillchen**
---
## Summary by cubic
Migrates shadcn UI from Radix primitives to Base UI across the app and adds migration docs. Fixes interaction regressions (tooltips, submenu triggers, switch roles) to stabilize the UI and e2e tests.

- **Migration**
  - Rewrote core UI in src/components/ui using Base UI (accordion, alert-dialog, button, checkbox, command, dialog, dropdown-menu, label, popover, scroll-area, select, separator, sheet, sidebar, switch, tabs, toggle, toggle-group, tooltip).
  - Updated 40+ components to new APIs (use polymorphic/as or direct elements instead of asChild, adjust Select onValueChange to handle undefined, align Accordion/Tabs props, replace data-[state] with data-open/checked, remove Tooltip wrappers around interactive elements, set submenu triggers to click/openOnHover=false, add aria-labels to switches/checkboxes, hide streaming animation text from a11y with aria-hidden).
  - Stabilized e2e tests (use click for submenu items, role-based locators for switches, waits for dialog close) and added .claude/run-e2e-update.sh for snapshot updates.
  - Added shadcn-migration.md documenting Radix→Base mappings and patterns.

- **Dependencies**
  - Added @base-ui/react and removed 16 @radix-ui/react-* packages.

<sup>Written for commit 043b0301d49ab72c1e020bbcddf395da94e7849a. Summary will update on new commits.</sup>

---

> [!NOTE]
> **Medium Risk**
> Broad UI-library swap plus many component/test updates; risk is mainly UI regressions and accessibility/interaction edge cases rather than backend/data/security changes.
> 
> **Overview**
> **Migrates shadcn UI primitives from Radix to Base UI** by adding `@base-ui/react`, removing most `@radix-ui/react-*` deps, and updating components to Base UI’s APIs (notably around `Dialog`, `DropdownMenu`, `Select`, `Switch`, triggers, and data attributes).
> 
> **Stabilizes UX + e2e coverage** by adjusting menu/submenu interactions (hover→click), adding explicit waits for dialogs to close, tightening selectors (e.g., role-based switches), scoping ambiguous button clicks, and refreshing Playwright snapshots; also adds `.claude/run-e2e-update.sh` to update snapshots consistently.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 043b0301d49ab72c1e020bbcddf395da94e7849a. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

---

<a href="https://app.devin.ai/review/dyad-sh/dyad/pull/2432">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open with Devin
  </picture>
</a>

### CODE REVIEW

**Comment by wwwillchen** [comment]
@BugBot run

**Comment by wwwillchen** [comment]
@BugBot run

**Comment by wwwillchen** [comment]
@BugBot run

**Comment by wwwillchen** [comment]
## ✅ Claude Code completed successfully

### Summary
- **Review comments**: 3 unresolved threads found, all previously flagged for human review (no new action needed)
- **CI fix**: Fixed 40 TypeScript type errors (TS7006: implicit `any` type) across 10 files by adding explicit type annotations to callback parameters

<details>
<summary>Details</summary>

### Review Comments (3 unresolved threads)
All 3 threads were already flagged for human review by a previous bot run with detailed options:
1. **`src/components/CreatePromptDialog.tsx`** (cursor) - Duplicated button styling across trigger components → already flagged with 3 resolution options
2. **`src/components/ui/tooltip.tsx`** (github-actions) - Tooltip creates nested TooltipProvider on every use → already flagged with 3 architectural options
3. **`src/components/ModelPicker.tsx`** (github-actions) - Nested TooltipTrigger > DropdownMenuTrigger creates invalid HTML → already flagged with 3 resolution options

### CI Fixes
**Root cause**: The `@base-ui/react` package was declared in `package.json` but not installed in `node_modules`, causing 16 module resolution errors. After `npm install`, 24 additional implicit `any` errors remained in callback parameters that needed explicit type annotations.

**Files fixed** (10 files, 19 insertions, 12 deletions):
- `ChatList.tsx` - `open: boolean` for DropdownMenu.onOpenChange
- `ForceCloseDialog.tsx` - `open: boolean` for AlertDialog.onOpenChange
- `GitHubConnector.tsx` - `value: string | null` for Select.onValueChange
- `GithubBranchManager.tsx` - `open: boolean` for AlertDialog/DropdownMenu.onOpenChange
- `ImportAppDialog.tsx` - `checked: boolean` for Checkbox.onCheckedChange
- `McpToolsPicker.tsx` - `v: string | null` for Select.onValueChange
- `ProviderSettings.tsx` - `open: boolean` for AlertDialog.onOpenChange
- `SupabaseConnector.tsx` - `supabaseBranchProjectId: string | null` for Select.onValueChange
- `chat/FileAttachmentDropdown.tsx` - `e: React.SyntheticEvent` for DropdownMenuItem.onSelect
- `preview_panel/SecurityPanel.tsx` - `e: React.MouseEvent` for Checkbox.onClick

**Verification**: All checks pass locally (lint ✅, formatting ✅, type check ✅, 661 unit tests ✅)

</details>

**Comment by wwwillchen** [comment]
@BugBot run

**Comment by wwwillchen** [comment]
@BugBot run

**Comment by wwwillchen** [comment]
@BugBot run

**Comment by wwwillchen** [comment]
@BugBot run

**Comment by wwwillchen** [comment]
@BugBot run

**Comment by wwwillchen** [comment]
@BugBot run

File: src/components/ui/alert-dialog.tsx [hunk 1]
```diff
@@ -119,9 +113,10 @@ function AlertDialogDescription({
 function AlertDialogAction({
   className,
   ...props
-}: React.ComponentProps<typeof AlertDialogPrimitive.Action>) {
+}: React.ComponentProps<"button">) {
   return (
-    <AlertDialogPrimitive.Action
+    <button
```
[the remark this answers is not in the document]
Comment (wwwillchen) [reply src/components/ui/alert-dialog.tsx]: This has already been addressed. `AlertDialogAction` wraps `AlertDialogPrimitive.Close` (see alert-dialog.tsx lines 113-124), which auto-closes the dialog on click. The migration correctly maps Action to Close since Base UI's AlertDialog doesn't have a separate Action component.

File: src/components/chat/FileAttachmentDropdown.tsx [hunk 2]
```diff
@@ -46,49 +40,33 @@ export function FileAttachmentDropdown({
...
-              Attach file as chat context
-            </DropdownMenuItem>
-          </TooltipTrigger>
-          <TooltipContent side="right">
-            Example use case: screenshot of the app to point out a UI issue
-          </TooltipContent>
-        </Tooltip>
-      </TooltipProvider>
+      <DropdownMenuItem
+        onSelect={(e: React.SyntheticEvent) => {
```
[the remark this answers is not in the document]
Comment (wwwillchen) [reply src/components/chat/FileAttachmentDropdown.tsx]: This has already been addressed. The current code uses `onClick` with `e.preventDefault()` (see FileAttachmentDropdown.tsx lines 44 and 58), not `onSelect`. The migration correctly uses the Base UI event handler pattern.

File: src/components/ImportAppDialog.tsx [hunk 3]
```diff
@@ -446,7 +442,7 @@ export function ImportAppDialog({ isOpen, onClose }: ImportAppDialogProps) {
                       </div>
                     </div>
 
-                    <Accordion type="single" collapsible>
+                    <Accordion>
```
[the remark this answers is not in the document]
Comment (wwwillchen) [reply src/components/ImportAppDialog.tsx:445]: This is already handled correctly. Base UI's Accordion defaults to `multiple: false` (single item at a time), and items are collapsible by default (clicking an expanded item collapses it via `value[0] === newValue ? [] : [newValue]` logic). This matches the previous Radix `type="single" collapsible` behavior without needing explicit props.

File: src/components/preview_panel/PreviewIframe.tsx [hunk 4]
```diff
@@ -1157,19 +1157,16 @@ export const PreviewIframe = ({ loading }: { loading: boolean }) => {
                   <MonitorSmartphone size={16} />
                 </button>
               </PopoverTrigger>
-              <PopoverContent
-                className="w-auto p-2"
-                onOpenAutoFocus={(e) => e.preventDefault()}
-                onInteractOutside={(e) => e.preventDefault()}
-              >
+              <PopoverContent className="w-auto p-2">
```
[the remark this answers is not in the document]
Comment (wwwillchen) [reply src/components/preview_panel/PreviewIframe.tsx:1129]: 🚩 **Flagged for human review**: The removal of `onInteractOutside={(e) => e.preventDefault()}` changes the device mode popover behavior — it will now close on outside clicks. This may be intentional to match Base UI defaults, but it does change the existing UX. A human should decide whether this behavior change is acceptable or if the popover should stay open on outside clicks (which would require implementing a custom solution with Base UI).

File: src/components/ui/toggle-group.tsx [hunk 5]
```diff
@@ -44,12 +44,11 @@ function ToggleGroupItem({
...
   size,
   ...props
-}: React.ComponentProps<typeof ToggleGroupPrimitive.Item> &
-  VariantProps<typeof toggleVariants>) {
+}: TogglePrimitive.Props & VariantProps<typeof toggleVariants>) {
   const context = React.useContext(ToggleGroupContext);
 
   return (
-    <ToggleGroupPrimitive.Item
+    <TogglePrimitive
```
[the remark this answers is not in the document]
Comment (wwwillchen) [reply src/components/ui/toggle-group.tsx:51]: 🚩 **Flagged for human review**: ToggleGroupItem uses a standalone `Toggle` component from `@base-ui/react/toggle` rather than a group-aware item. Base UI's `ToggleGroup` manages state via its `value`/`onValueChange` props, and the standalone `Toggle` does participate in the group's state when rendered as a child of `ToggleGroup`. However, this should be verified in practice to ensure proper state synchronization.

File: src/components/preview_panel/PreviewIframe.tsx [hunk 6]
```diff
@@ -1139,83 +1110,59 @@ export const PreviewIframe = ({ loading }: { loading: boolean }) => {
...
+                  "p-1 rounded hover:bg-gray-200 dark:hover:bg-gray-700 dark:text-gray-300",
+                  deviceMode !== "desktop" && "bg-gray-200 dark:bg-gray-700",
+                )}
+                title="Device Mode"
+              >
+                <MonitorSmartphone size={16} />
+              </PopoverTrigger>
+              <PopoverContent className="w-auto p-2">
+                <ToggleGroup
+                  value={[deviceMode]}
```
[the remark this answers is not in the document]
Comment (wwwillchen) [reply src/components/preview_panel/PreviewIframe.tsx:1131]: 🚩 **Flagged for human review**: The ToggleGroup single-select emulation using `value={[deviceMode]}` and extracting via `value[value.length - 1]` works but is fragile. Base UI's ToggleGroup is multi-select by default, and clicking the active item may produce an empty array. The current code handles this with the `value.length > 0` check, but a human should verify the UX is correct — especially whether deselecting the active device mode (producing no selection) is acceptable.

File: src/components/ui/dropdown-menu.tsx [hunk 7]
```diff
@@ -190,49 +185,66 @@ function DropdownMenuShortcut({
...
-}: React.ComponentProps<typeof DropdownMenuPrimitive.SubTrigger> & {
+}: MenuPrimitive.SubmenuTrigger.Props & {
   inset?: boolean;
 }) {
   return (
-    <DropdownMenuPrimitive.SubTrigger
+    <MenuPrimitive.SubmenuTrigger
       data-slot="dropdown-menu-sub-trigger"
       data-inset={inset}
+      openOnHover={false}
```
[the remark this answers is not in the document]
Comment (wwwillchen) [reply src/components/ui/dropdown-menu.tsx:204]: 🚩 **Flagged for human review**: `openOnHover={false}` was set intentionally to make submenus click-to-open instead of hover-to-open. This was done to stabilize e2e tests and address interaction issues. A human should decide whether this UX trade-off is acceptable for production, or if hover-to-open should be re-enabled (which may require addressing the underlying interaction issues).

File: src/components/ProModeSelector.tsx [hunk 8]
```diff
@@ -218,76 +195,46 @@ function TurboEditsSelector({
...
-              </li>
-            </ul>
-          </TooltipContent>
-        </Tooltip>
+        <span title="Edits files efficiently without full rewrites. Classic: Uses a smaller model to complete edits. Search & replace: Find and replaces specific text blocks.">
+          <Info
+            className={`h-4 w-4 cursor-help ${!isTogglable ? "text-muted-foreground/50" : "text-muted-foreground"}`}
+          />
+        </span>
       </div>
```
[the remark this answers is not in the document]
Comment (wwwillchen) [reply src/components/ProModeSelector.tsx:203]: 🚩 **Flagged for human review**: The rich HTML tooltip content for Turbo Edits was replaced with a plain `title` attribute, losing formatting (bold, lists, line breaks). A human should decide whether to restore a proper `Tooltip` + `TooltipContent` here since this is a place where rich content genuinely helps users understand the modes.

File: src/components/chat/TokenBar.tsx [hunk 9]
```diff
@@ -54,7 +54,7 @@ export function TokenBar({ chatId }: TokenBarProps) {
     <div className="px-4 pb-2 text-xs" data-testid="token-bar">
       <TooltipProvider>
         <Tooltip>
-          <TooltipTrigger asChild>
+          <TooltipTrigger>
```
[the remark this answers is not in the document]
Comment (wwwillchen) [reply src/components/chat/TokenBar.tsx:57]: 🚩 **Flagged for human review**: Base UI's `Tooltip.Trigger` renders as a `<button>` by default. Without `asChild` or a `render` prop, wrapping a `<div>` creates a block element inside a button. This affects several files (TokenBar, ChatHeader, etc.). A human should decide whether to use the `render` prop pattern (e.g., `<TooltipTrigger render={<div />}>`) to avoid the button semantics on non-interactive elements.

**Comment by wwwillchen** [comment]
@BugBot run

**Comment by wwwillchen** [comment]
@BugBot run

**Comment by wwwillchen** [comment]
@BugBot run
